autosklearn/automl.py

```diff
@@ -1074,30 +1052,59 @@ def fit_pipeline(
                 instance=None,
                 instance_specific=None,
                 seed=self._seed,
-                cutoff=cutoff,
+                cutoff=kwargs.pop('cutoff', self._per_run_time_limit),
                 capped=False,
             )
         )
 
         pipeline = None
-        if run_value.status == StatusType.SUCCESS:
+        if kwargs['disable_file_output'] or kwargs['resampling_strategy'] == 'test':
+            self._logger.warning("File output is disabled. No pipeline can returned")
+        elif run_value.status == StatusType.SUCCESS:
             if kwargs['resampling_strategy'] in ('cv', 'cv-iterative-fit'):
                 load_function = self._backend.load_cv_model_by_seed_and_id_and_budget
             else:
                 load_function = self._backend.load_model_by_seed_and_id_and_budget
-            try:
-                pipeline = load_function(
-                    seed=self._seed,
-                    idx=run_info.config.config_id + 1,
-                    budget=run_info.budget,
-                )
-            except Exception as e:
-                self._logger.warning(f"Cannot load pipeline because of {e}")
+            pipeline = load_function(
+                seed=self._seed,
+                idx=run_info.config.config_id + 1,
+                budget=run_info.budget,
+            )
 
         self._clean_logger()
 
         return pipeline, run_info, run_value
 
+    def _get_include_exclude_pipeline_dicts(self):
```

Comment:
This is now identical to parts of the code in `autosklearn.util.pipeline.get_configuration_space`, isn't it?

Reply:
Thanks for pointing this out, several places in the code had this. I have unified them all.